pkg/test/util/connection/connection.go

```diff
@@ -58,3 +67,38 @@ func CheckConnection(t *testing.T, conn Connection) error {
 	}
 	return nil
 }
+
+// CheckRBACRequest checks if a request is successful under RBAC policies.
+// Under RBAC policies, a request is consider successful if:
+// * If the policy is deny:
+// *** For HTTP: response code is 403
+// *** For TCP: EOF error
+// * If the policy is allow:
+// *** Response code is 200
+func CheckRBACRequest(t *testing.T, conn Connection) error {
```

Comment:
This is RBAC specific (as well as those consts and `IsAllowed`) and should be placed in the rbac_v2_test.go

Reply:
I put it there because in the future we may have more than one rbac tests and they can use/share this function. I can put it back to `rbac_v2_test.go` and refactor it later if we introduce more rbac test files.

Reply:
What prevents us to reuse it if it's in `rbac_v2_test.go` (or `util.go` under the rbac directory)?

The point is `connection/connection.go` is the test framework and it should be decoupled with the actual tests. If every test adds their `CheckXXXRequest()` to this file, then it will become very hard to maintain and use. Also note the `CheckRBACRequest()` is going to be only used by RBAC so putting it in `connection.go` cannot be shared with other test cases other than RBAC.

Reply:
I moved it to our rbac test file. However, I'm keeping constants like tcp port, http resp codes in the framework code.